src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -351,9 +351,39 @@ private static boolean containsAllSafeTokens(final DetailAST expr) {
             }
         }
 
-        return argIsNotNull
-                || !arg.branchContains(TokenTypes.IDENT)
-                    && !arg.branchContains(TokenTypes.LITERAL_NULL);
+        return argIsNotNull || !branchContainsIdentOrNull(arg);
+    }
+
+    /**
+     * Checks whether a {@code null} or {@code ident} exist under token.
+     *
+     * @param token tree token.
+     * @return true if a {@code null} or {@code ident} exists under token.
+     */
+    private static boolean branchContainsIdentOrNull(DetailAST token) {
```

Comment:
This is basically just a re-hash of the original `branchContains` method. There is no other way to look at what was intended and rewrite in some other fashion?
Like what are we looking for by `IDENT`? A variable, field, methods, etc or any of the above?

Reply:
You are right. This method is redundant here. To make the long story short, is handles 
```
        s.equalsIgnoreCase(((s = "cold pizza")));
```
but this should be done in the method `skipVariableAssign`.